### ISSUE TITLE
Playback issues after commit 062ae33

### ISSUE THREAD

**Issue by smp79** [xbmc/xbmc#14877]
The issue: infinite spinning circle when trying to play a video file during the playback of another file.
~~I can only reproduce this on Raspberry Pi.~~
How to reproduce: 
1. Play [Sample 1](https://www.dropbox.com/s/wucckb1j52w7eyr/50i-MBAFF.ts?dl=1). 
2. During the playback of Sample 1 start the playback of [Sample 2](https://www.dropbox.com/s/aypt4ejky3sfv0y/bd_sample.m2ts?dl=1).

Reverting 062ae33 fixes the issue.

[Debug log](https://paste.kodi.tv/ivenewihow.kodi)

**Comment by da-anda** [comment]
@peak3d since your commit has been identified as source of the issue, mind having a look?

@smp79 please stick to our issue template for your next ticket, thanks.

**Comment by peak3d** [comment]
@da-anda already looked into one of those, Buffer mamageent on rbpi seems to have an issue, @popcornmix could be the right person to investigate.

**Comment by popcornmix** [comment]
Not able to reproduce the problem either by launching playback through json (texturecache play) or playing from files menu, then back-spacing to gui and playing second file while first plays.

I'm using mmal and have "adjust display refresh rate" enabled (I assume problem doesn't occur without that setting?)

**Comment by smp79** [comment]
The problem does not occur when "adjust display refresh rate" is off. Here's a [video](https://youtu.be/l8wRWU1fSI4) of how I reproduce it.

**Comment by smp79** [comment]
This is not an RPi-specific issue, as it turns out. I just tested on a Windows PC with "Delay after change of refresh rate" set to 3 seconds and I got an infinite spinning circle.

**Comment by popcornmix** [comment]
I can confirm I see the failure when "Delay after change of refresh rate" is set to 3 seconds.
As it's not a pi specific issue I'll leave it to @peak3d to investigate.

**Comment by FernetMenta** [comment]
can you try adding code here: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L5019

```
void CVideoPlayer::OnResetDisplay()
{
  if (!m_displayLost)
    return
```

**Comment by FernetMenta** [comment]
next round:

```
diff --git a/xbmc/cores/VideoPlayer/VideoPlayer.cpp b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
index 01b4679670..9469f69081 100644
--- a/xbmc/cores/VideoPlayer/VideoPlayer.cpp
+++ b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
@@ -3693,6 +3693,8 @@ bool CVideoPlayer::OpenAudioStream(CDVDStreamInfo& hint, bool reset)
     if (!player->OpenStream(hint))
       return false;
 
+    player->SendMessage(new CDVDMsgBool(CDVDMsg::GENERAL_PAUSE, m_displayLost), 1);
+
     static_cast<IDVDStreamPlayerAudio*>(player)->SetSpeed(m_streamPlayerSpeed);
     m_CurrentAudio.syncState = IDVDStreamPlayer::SYNC_STARTING;
     m_CurrentAudio.packets = 0;
@@ -3782,6 +3784,8 @@ bool CVideoPlayer::OpenVideoStream(CDVDStreamInfo& hint, bool reset)
     if (!player->OpenStream(hint))
       return false;
 
+    player->SendMessage(new CDVDMsgBool(CDVDMsg::GENERAL_PAUSE, m_displayLost), 1);
+
     // look for any EDL files
     m_Edl.Clear();
     if (hint.fpsrate > 0 && hint.fpsscale > 0)
@@ -5034,6 +5038,9 @@ void CVideoPlayer::OnLostDisplay()
 
 void CVideoPlayer::OnResetDisplay()
 {
+  if (!m_displayLost)
+    return;
+
   CLog::Log(LOGNOTICE, "VideoPlayer: OnResetDisplay received");
   m_VideoPlayerAudio->SendMessage(new CDVDMsgBool(CDVDMsg::GENERAL_PAUSE, false), 1);
   m_VideoPlayerVideo->SendMessage(new CDVDMsgBool(CDVDMsg::GENERAL_PAUSE, false), 1);
```

**Comment by smp79** [comment]
> next round:

That fixed it.

**Comment by a1rwulf** [comment]
@FernetMenta what does this patch do exactly?
I read it as:
When opening a stream while we have no display, we pause the player.
But how long do we pause and when/where is it resumed?
The early return in `OnResetDisplay` means we do nothing when we actually didn't lose display?
How can this happen at all?

**Comment by FernetMenta** [comment]
The stream player's message queue is kind of old fashioned. It only delivers messages if a stream is active. But communication also occurs during times no stream is active. Those messages were not delivered due to inactive msg queue.
As a result OnLostDisplay paused video player, video stream was closed due to missing sps, OnResetDisplay could not unpause VPV. Now demuxer split sps, VP opened new video stream but VPV was still in pause.

My change synchronises state not related to stream when message queue is active again. This issue wouldn't have happened with proper messaging seen in ActiveAE, VDPAU, or VAAPI.

**Comment by vbier** [comment]
Is this related to #14867, which also complains about the spinning circle not going away when starting recordings?

**Comment by andrewgreenwood** [comment]
@vbier I wondered this when I first saw this issue but since it appears to depend on having "adjust display refresh rate" enabled (one of the comments mentions this, and the code changed in the specified commit only gets executed if CSettings::SETTING_VIDEOPLAYER_ADJUSTREFRESHRATE is not ADJUST_REFRESHRATE_OFF) I figured it was likely something else.

Still, it could be the same underlying problem with different triggers.

For this particular issue, does the video still play but with no on-screen controls enabled and a circle over the top of the video? (As that's what I experience in my issue)
